autopilot/contractor/contractor.go

```diff
@@ -1222,67 +1084,21 @@ func performContractMaintenance(ctx *mCtx, alerter alerts.Alerter, bus Bus, chur
 		logger:          logger.Named("ipFilter"),
 		subnetToHostKey: make(map[string]string),
 	}
-	keptContracts, churnReasons, err := performContractChecks(ctx, alerter, bus, cc, cr, ipFilter, logger)
+	nGood, nUpdated, err := performContractChecks(ctx, alerter, bus, churn, cc, cr, ipFilter, logger)
```

Comment:
You shouldn't need `nGood` anymore. `performContractFormations` can fetch the good contracts and then compute the number of `wanted` contracts based off of that. That also means it doesn't require `wanted` as an input anymore.
That is more consistent with the host checks. Where we also update the hosts in one function and instead of returning the good hosts we just fetch them again later.